File: LeetCode/p07_Reverse_Number/main.cpp
#include<bits/stdc++.h>

using namespace std;

class Solution{
public:
    int reverse(int n){
        int rev = 0;

        while(n != 0){
            int digit = n % 10;
            n /= 10;

            if(rev > INT_MAX / 10 || (rev == INT_MAX && digit > 7)){
                return 0;
            }

            if(rev < INT_MIN / 10 || (rev == INT_MIN && digit < -8)){
                return 0;
            }

            rev = (rev * 10) + digit;
        }

        return rev;
    }
};

File: LeetCode/p07_Reverse_Number/main.ts
class Solution{
    reverse(n: number){
        let rev = 0;

        const INT_MAX = 2147483647;
        const INT_MIN = -2147483648;

        while(n != 0){
            let digit = n % 10;
            n = Math.trunc(n / 10);

            if(rev > Math.trunc(INT_MAX / 10) || (rev == Math.trunc(INT_MAX / 10) && digit > 7)){
                return 0;
            }

            if(rev > Math.trunc(INT_MIN / 10) || (rev == Math.trunc(INT_MIN / 10) && digit < -8)){
                return 0;
            }

            rev = (rev * 10) + digit;
        }

        return rev;
    }
}